internal/tui/tui.go

```diff
@@ -214,9 +225,12 @@ func (a *appModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 			return a, nil
 		}
 
-		// forward to page
+		// Forward to view.
 		updated, itemCmd := item.Update(msg)
-		a.pages[a.currentPage] = updated.(util.Model)
+		if model, ok := updated.(util.Model); ok {
+			a.pages[a.currentPage] = model
```

Comment:
does `a.pages` get concurrent access?

if so, could be a `csync.Map` 🤔

Reply:
Good question: I don't know. @kujtimiihoxha?

If it is needed, @caarlos0 could you update accordingly?